### DISCUSSION TITLE
<Environment /> loading slowly in react-native

### DISCUSSION THREAD

**Discussion by jjo10** [pmndrs/drei#1727] [Q&A]
three version: 0.158.0
@react-three/fiber version: 8.15.11
@react-three/drei version: 9.88.16

I am finding that my react-native app actually freezes briefly when rendering either <Environment /> or <Stage />. It is running on a real Samsung Galaxy S21 Ultra. Currently I'm dealing with this by using a fallback component with React.Suspense. I've tried presets and a plain white .hdr (~200kB) and it's still slow to load + UI freezes briefly. 

Are there any ways to improve performance in this case?

**Comment by ninjz** [comment]
How are you able to get environments working in react-native? I'm unable to get any to load... Always returns that .hdr file is undefined.
